### PR TITLE
port two-libs sample to gradle

### LINKED ISSUE #113: add sample for 0.4.0 dependencies management
/cc @Whiteseeker

### PR DESCRIPTION
**Pull request by proppy**
This show how to use the new 0.3.0 dependencies support.

I'd also like to show in the same sample how to depends on a .so and add it in the .apk, maybe the name 'two-libs' is not really relevant anymore.

Any thoughts?

/cc @rschiu @ggfan @Whiteseeker @ph0b 

Fixes #113 #57

### CODE REVIEW

**Comment by dragostis** [comment]
This is really similar to what I want to achieve apart from the fact that the library doesn't support Java files.

Are the normal libraries also getting the `productFlavor` [options](http://stackoverflow.com/questions/32957274/add-productflavor-to-experimental-android-gradle-plugin-library)?

File: two-libs/lib/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apply plugin: "com.android.model.native"
+
+model {
+    android {
+        compileSdkVersion = 23
+        buildToolsVersion = "23.0.0"
```
Comment (rschiu) [inline two-libs/lib/build.gradle]: Only the compileSdkVersion is needed.  Anything else in the android{} block should be removed.

File: two-libs/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        buildToolsVersion = "23.0.0"
+
+        defaultConfig.with {
+            applicationId = "com.example.twolibs"
+            minSdkVersion.apiLevel = 4
+            targetSdkVersion.apiLevel = 23
+        }
+    }
+
+    compileOptions.with {
```
Comment (rschiu) [inline two-libs/app/build.gradle]: Do we still need the compileOptions?

Comment (proppy) [inline two-libs/app/build.gradle]: Yes, otherwise it would failed if java8 is installed.

**Comment by proppy** [comment]
I think we should refactor this into a proper `native library sample` rather naively porting `two-libs`.

@claywilkinson suggest the following a gradle project with 3 modules.
- a `NativeLibrary` module
- 1 `AppDynamicLibrary` module that depends on `NativeLibrary` with dynamic linkage
- 1 `AppStaticLibrary` module that depends on `NativeLibrary` with static linkage

**Comment by koyach** [comment]
Yeah sounds good

**Comment by proppy** [comment]
`hello-libs` does this and more now.

**Pull request closed without merging**